File: Examples/android/CMeSdkJava/app_rework/src/main/java/me/connect/sdk/java/sample/db/dao/ProofRequestDao.java
package me.connect.sdk.java.sample.db.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import me.connect.sdk.java.sample.db.entity.ProofRequest;

@Dao
public interface ProofRequestDao {
    @Query("SELECT * FROM proofrequest")
    List<ProofRequest> getAll();

    @Query("SELECT * FROM proofrequest where id = :id")
    ProofRequest getById(int id);

    @Query("SELECT EXISTS(SELECT * FROM proofrequest WHERE thread_id = :threadId)")
    boolean checkExists(String threadId);

    @Insert
    void insertAll(ProofRequest... proofRequests);

    @Update
    void update(ProofRequest proofRequest);

    @Delete
    void delete(ProofRequest proofRequest);
}

File: Examples/android/CMeSdkJava/app_kt/src/main/kotlin/me/connect/sdk/java/samplekt/db/dao/ProofRequestDao.kt
package me.connect.sdk.java.samplekt.db.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import me.connect.sdk.java.samplekt.db.entity.ProofRequest


@Dao
interface ProofRequestDao {
    @Query("SELECT * FROM proofrequest")
    fun getAll(): LiveData<List<ProofRequest>>

    @Query("SELECT * FROM proofrequest where id = :id")
    suspend fun getById(id: Int): ProofRequest

    @Query("SELECT EXISTS(SELECT * FROM proofrequest WHERE thread_id = :threadId)")
    suspend fun checkExists(threadId: String): Boolean

    @Insert
    suspend fun insertAll(vararg proofRequests: ProofRequest)

    @Update
    suspend fun update(proofRequest: ProofRequest)

    @Delete
    suspend fun delete(proofRequest: ProofRequest)
}